packages/fxa-payments-server/src/components/PlanDetails/index.tsx

```diff
@@ -58,18 +68,72 @@ export const PlanDetails = ({
     ? { background: webIconBackground }
     : '';
 
-  const discountAmount =
-    coupon && amount && coupon.discountAmount
-      ? amount - coupon.discountAmount
-      : amount;
+  const infoBoxMessage = useInfoBoxMessage(coupon, selectedPlan);
+
   const planPrice = formatPlanPricing(
-    discountAmount,
+    amount,
     currency,
     interval,
     interval_count
   );
 
-  const infoBoxMessage = useInfoBoxMessage(coupon, selectedPlan);
+  useEffect(() => {
+    async function fetchData() {
+      setLoading(true);
+      const invoicePreview = await apiInvoicePreview({
+        priceId: selectedPlan.plan_id,
+        promotionCode: coupon?.promotionCode as string,
+      });
```

Comment:
Q: is there any chance that `apiInvoicePreview()` can return `undefined` or throw an error; like: https://github.com/mozilla/fxa/blob/de4a24a6b0d0c4a2b0cad08bf07d3d24b0320c36/packages/fxa-payments-server/src/lib/apiClient.ts#L106

Reply:
looking into this

Reply:
yes, good catch, and because of the fetchData call being async its difficult to catch this error if it happens and was resulting in a stuck loading spinner.

ive switch the logic around a little bit to account for this and to fail gracefully in a way that displays what data we do have.